#!/usr/bin/env python

PACKAGE = 'neuronbot2_bringup'

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, str_t, double_t, bool_t, int_t

gen = ParameterGenerator()

gen.add("wheel_diameter", int_t, 0, "The diameter of the wheel", 84, 10, 500)
gen.add("wheel_track", int_t, 0, "The track of the wheel", 225, 50, 1000)
gen.add("encoder_resolution", int_t, 0, "The resolution of the encoder", 1428, 100 , 32000)

gen.add("do_pid_interval", int_t, 0, "The interval of do pid", 10, 1, 80)
gen.add("kp", int_t, 0, "Kp value", 75, 0, 10000)
gen.add("ki", int_t, 0, "Ki value", 2500, 0, 32000)
gen.add("kd", int_t, 0, "Kd value", 0, 0, 1000)
gen.add("ko", int_t, 0, "Ko value", 10, 0, 1000)

gen.add("cmd_last_time", int_t, 0, "cmd_last_time value", 250, 0, 1000)

gen.add("max_v_liner_x", int_t, 0, "liner x", 40, 0, 500)
gen.add("max_v_liner_y", int_t, 0, "liner y", 0, 0, 500)
gen.add("max_v_angular_z", int_t, 0, "angular z", 150, 0, 2000)

gen.add("imu_type", int_t, 0, "imu type(`E`(69) for enable", 69, 0, 255)

exit(gen.generate(PACKAGE, "neuronbot2_bringup", "neuronbot2_driver"))
